src/v/utils/variant.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include <variant>
+
+namespace {
```

Comment:
what's the purpose of an anonymous namespace in a header?

Reply:
I separated it into a reusable header only as a very last stage and didn't think about it. I'll change it, thank you.